hibernate-core/src/test/java/org/hibernate/jpa/test/query/QueryTest.java

```diff
@@ -704,6 +704,138 @@ public void testJpaPositionalParameters() {
 			em.close();
 		}
 	}
+	
+	/**
+	 * Collection parameters are internally rewritten to named parameters (one named parameter by
+	 * collection item) and HQL parser rejects mixed positional and named parameters queries.
+	 */
+	@Test
+	@TestForIssue(jiraKey = "HHH-12290")
+	public void testParameterCollectionAndPositional() {
+		final Item item = new Item( "Mouse", "Micro$oft mouse" );
+		final Item item2 = new Item( "Computer", "Dell computer" );
+
+		EntityManager em = getOrCreateEntityManager();
+		em.getTransaction().begin();
+		try {
+			em.persist( item );
+			em.persist( item2 );
+			assertTrue( em.contains( item ) );
+			em.getTransaction().commit();
+
+			em.getTransaction().begin();
+			Query q = em.createQuery( "select item from Item item where item.name in ?1 and item.descr = ?2" );
+			//test hint in value and string
+			q.setHint( "org.hibernate.fetchSize", 10 );
+			q.setHint( "org.hibernate.fetchSize", "10" );
```

Comment:
I don't think these hints have any value here? I think they should be removed to simplify the test case.